public/app/features/admin/ldap/LdapSettingsPage.tsx

```diff
@@ -0,0 +1,315 @@
+          group_search_base_dns: [],
+          group_search_filter: '',
+          group_search_filter_user_attribute: '',
+          host: '',
+          min_tls_version: '',
+          port: 389,
+          root_ca_cert: '',
+          search_base_dns: [],
+          search_filter: '',
+          skip_org_role_sync: false,
+          ssl_skip_verify: false,
+          start_tls: false,
+          timeout: 10,
+          tls_ciphers: [],
+          tls_skip_verify: false,
+          use_ssl: false,
+        },
+      ],
+    },
+    enabled: false,
+    skipOrgRoleSync: false,
+    syncCron: '',
+  },
+};
+
+export const LdapSettingsPage = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [formSettings, setFormSettings] = useState<LdapPayload>();
+
+  const methods = useForm<LdapPayload>({ defaultValues: emptySettings });
+  const { handleSubmit, register, reset } = methods;
+
+  const styles = useStyles2(getStyles);
+
+  useEffect(() => {
+    async function init() {
+      const payload = await getBackendSrv().get<LdapPayload>('/api/v1/sso-settings/ldap');
+      if (!payload || !payload.settings || !payload.settings.config) {
+        appEvents.publish({
+          type: AppEvents.alertError.name,
+          payload: [t('ldap-settings-page.alert.error-fetching', 'Error fetching LDAP settings')],
+        });
+        return;
+      }
+
+      setFormSettings(payload);
+      setIsLoading(false);
+    }
+    init();
+  }, []);
+
+  useEffect(() => {
+    reset(formSettings);
+  }, [reset, formSettings]);
+
+  /**
+   * Display warning if the feature flag is disabled
+   */
+  if (!config.featureToggles.ssoSettingsLDAP) {
+    return (
+      <Alert title="invalid configuration">
+        <Trans i18nKey="ldap-settings-page.alert.feature-flag-disabled">
+          This page is only accessible by enabling the <strong>ssoSettingsLDAP</strong> feature flag.
+        </Trans>
+      </Alert>
+    );
+  }
+
+  const submitAndEnableLdapSettings = async (fm: LdapPayload) => {
+    try {
+      const result = await getBackendSrv().put('/api/v1/sso-settings/ldap', fm);
+      if (result) {
+        appEvents.publish({
+          type: AppEvents.alertError.name,
+          payload: [t('ldap-settings-page.alert.error-saving', 'Error saving LDAP settings')],
+        });
+      }
+      appEvents.publish({
+        type: AppEvents.alertSuccess.name,
+        payload: [t('ldap-settings-page.alert.saved', 'LDAP settings saved')],
+      });
+    } catch (error) {
+      appEvents.publish({
+        type: AppEvents.alertError.name,
+        payload: [t('ldap-settings-page.alert.error-saving', 'Error saving LDAP settings')],
+      });
+    }
+  };
+
+  const onErrors = () => {
+    appEvents.publish({
+      type: AppEvents.alertError.name,
+      payload: [t('ldap-settings-page.alert.error-validate-form', 'Error validating LDAP settings')],
+    });
+  };
+
+  /**
+   * Button's Actions
+   */
+  const saveForm = async () => {
```

Comment:
I'm confused what's the difference between save and enable and just save? Both actions make the call to the same endpoint but with different data. The both functions are similar enough to be combined into one with a different payload argument. 

Also not sure about the role of `formSettings`. It looks like this is the data fetched from backend and used to populate the form, so why is it used for saving here?

Reply:
I refactor this to be more clear. Persiting to backend is now an external function and the difference between the 2 functions is enabling the service